packages/embark-library-manager/src/npm.js

```diff
@@ -1,35 +1,64 @@
-const fs = require('fs');
-const PluginManager = require('live-plugin-manager-git-fix').PluginManager;
+const path = require('path');
 require('colors');
 import { LongRunningProcessTimer } from 'embark-utils';
+const {exec} = require('child_process');
+const {existsSync, mkdirpSync, writeFileSync} = require('fs-extra');
+const rimraf = require('rimraf');
 
 class Npm {
 
   constructor(options) {
     this._logger = options.logger;
     this._packageName = options.packageName;
     this._version = options.version;
+    this._installed = {};
     this._installing = {};
+    this._installFailed = {};
     this._useDashboard = options.useDashboard;
   }
 
   static getPackagePath(packageName, version) {
-    return './.embark/versions/' + packageName + '/' + version + '/' + packageName;
+    return path.join('.embark', 'versions', packageName, version, 'node_modules', packageName);
+  }
+
+  _isInstallFailure(packageName, version) {
+    return !!this._installFailed[packageName + version];
+  }
+
+  _isInstalled(packageName, version) {
+    if (this._installed[packageName + version]) {
+      return true;
+    }
+    const packagePath = Npm.getPackagePath(packageName, version);
+    const installPath = path.dirname(path.dirname(packagePath));
+    if (existsSync(packagePath) &&
+        !this._isInstalling(packageName, version)) {
+      if (existsSync(path.join(installPath, 'finished'))) {
```

Comment:
I assume this `finished` path is to ensure we have a path to check that is indeed completed installing? Cause theoretically npm install could break in mid-air but might have created the original path already. 

If the above is true, I wonder if we can make `finished` a bit better, either by comments or by putting it in a constant to give it more semantic meaning.

What do you think?

Reply:
Your interpretation is correct. I'll use a constant.